### DISCUSSION TITLE
Error on reading the abaqus file (.inp)

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20903] [Q&A Modules: General] [answered]
Hello MOOSE Experts

I generated a ABAQUS mesh input file (.inp) using CUBIT with several grains. After I am trying to read the file in the MOOSE as follows:

```
[Mesh]
  [initial_mesh]
     type = FileMeshGenerator
     file = Abaqus_file.inp
  []
    [./scale]
    type = TransformGenerator
    input = initial_mesh
    transform = SCALE
    vector_value ='11.0e-6 11.0e-6 0'
  []
   construct_side_list_from_node_list=true
[]
```

Afterwards, I am running the command with --mesh-only option and getting the following error:

```
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'
/home/asingh/projects/babbler/babbler-opt: Relink `/home/asingh/mambaforge3/envs/moose/lib/libgfortran.so.5' with `/lib64/librt.so.1' for IFUNC symbol `clock_gettime'



*** Warning, This code is deprecated and will be removed in future versions:
The parameter direction is deprecated.
Specifying direction+multiapp is deprecated. Specify the to_multi_app and from_multi_app

Error: Needed to read 2 nodes, but read 4 instead!
[0] ../src/mesh/abaqus_io.C, line 723, compiled Apr 13 2022 at 21:08:08


*** ERROR ***
Error: Needed to read 2 nodes, but read 4 instead!

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

If the number of grains are less than 30, it is working absolutely fine. However, if the number of grains increases even to 31, the above mentioned error is popping out.

Please give me any suggestions to solve the issue. I have attached the mesh file as well.
[mesh_file.zip](https://github.com/idaholab/moose/files/8584403/mesh_file.zip)

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

For the first errors, I'd remove the moose environment in mamba and re-install. Something changed in the libgfortran library since it was compiled, and easiest way is to reinstall. 

For the deprecation warning, your multiapp syntax is deprecated. Instead of
```
 direction = TO_MULTIAPP
  multiapp = multiapp_name
```
we now do
```
to_multiapp = multiapp_name
```
this is to support transfers from multiapp to multiapp. 

What's a good reader for abaqus meshes? I dont have Abaqus on my machine. 

Guillaume

**Reply by avtarsinghh1991** [reply]
Hi @GiudGiud 

Thanks. I solved the multiapp issue. 

For the abaqus file, I tried this file to read in ABAQUS, its working absolutely fine in ABAQUS.

Best
Avtar

**Reply by GiudGiud** [reply]
Well the error isnt explicit enough. My guess is that there are mixed element types in a domain, like a single triangle in a quad block. 
MOOSE doesnt support this. 

I converted it to exodus with meshio. Do you get a better error message with this format?
[mesh_file_exo.zip](https://github.com/idaholab/moose/files/8584755/mesh_file_exo.zip)

**Reply by avtarsinghh1991** [reply]
@GiudGiud 

Thanks for the insight. I have also written the abaqus input using my own MATLAB subroutine. For that I am generating structured mesh in which there is no possibility of mixed element type.

I am facing the same issue in there as well with more than 30 grains.

I am getting the following error upon reading the file you shared in MOOSE

```
EXODUS: Error: Attempting to open the netcdf-4 file:
        '/home/singh/projects/babbler/Anisotropic_2D/Abaqus_file.e'
        with a netcdf library that does not support netcdf-4
Error opening ExodusII mesh file: /home/singh/projects/babbler/Anisotropic_2D/Abaqus_file.e
[0] ../src/mesh/exodusII_io_helper.C, line 643, compiled Dec 16 2021 at 14:15:31


*** ERROR ***
Error opening ExodusII mesh file: /home/singh/projects/babbler/Anisotropic_2D/Abaqus_file.e

```


I have tried importing the abaqus file in cubit and export it as exodus file. It is working fine if doing so. But i want to directly use the abaqus file. Any suggestions?

Best
Avtar

**Reply by GiudGiud** [reply]
ok you cant read my file because I m using an hdf5 exodus and I dont think your moose install is setup that way. 

Only suggestion I would have is to debug the abaqus file reader in MOOSE. Ill create an issue and the thermomechanics people at INL may look at it if they have time

**Reply by avtarsinghh1991** [reply]
Thanks. Can you suggest any other file format which I can write including the nodesets and sidesets? I am thinking of using the .msh file format

**Reply by GiudGiud** [reply]
The list of supported (well except for inp in your case) is here:
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html
